kion/resource_global_permission_mapping.go

```diff
@@ -0,0 +1,197 @@
+package kion
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	hc "github.com/kionsoftware/terraform-provider-kion/kion/internal/kionclient"
+)
+
+func resourceGlobalPermissionMapping() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceGlobalPermissionMappingCreate,
+		ReadContext:   resourceGlobalPermissionMappingRead,
+		UpdateContext: resourceGlobalPermissionMappingUpdate,
+		DeleteContext: resourceGlobalPermissionMappingDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGlobalPermissionMappingImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"app_role_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Application role ID for the permission mapping.",
+			},
+			"user_groups_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+				Description: "Set of user group IDs for the permission mapping.",
+			},
+			"user_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+				Description: "Set of user IDs for the permission mapping.",
+			},
+		},
+	}
+}
+
+func resourceGlobalPermissionMappingCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*hc.Client)
+
+	appRoleID := d.Get("app_role_id").(int)
+
+	mapping := hc.GlobalPermissionMapping{
+		AppRoleID:     appRoleID,
+		UserGroupsIDs: hc.ConvertInterfaceSliceToIntSlice(d.Get("user_groups_ids").(*schema.Set).List()),
+		UserIDs:       hc.ConvertInterfaceSliceToIntSlice(d.Get("user_ids").(*schema.Set).List()),
+	}
+
+	_, err := client.POST("/v3/global/permission-mapping", []hc.GlobalPermissionMapping{mapping})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("global-%d", appRoleID))
```

Comment:
Global is implied by the resource type.  I'm not sure you need to use a composite id here, especially since you never seem to enforce that `global-` is passed anywhere else (aka you only use what's after the hyphen elsewhere in code).  AppRoleID is probably enough.  Thoughts?

Reply:
It's used in `terraform import,` but I suppose that's unnecessary because we define the resource type during the `import.`  I can look at changing this.